src/hooks/useOnce.ts

```diff
@@ -0,0 +1,16 @@
+import { useRef } from 'react'
+
+export function useOnce() {
+  const done = useRef(false)
+
+  const once = {
+    done,
+    do: (effect: React.EffectCallback) => {
+      if (done.current) return
+      effect()
+      done.current = true
```

Comment:
`React.EffectCallback` can return a cleaning function to be called when a component is unmounted

```suggestion
      if (done.current) return
      done.current = true
      return effect()
```